### PR TITLE
Have a way to remap machine class to fdb tier

### PR DESCRIPTION
**Pull request by riverszhang89**
The new "integration" tier is an alpha environment but runs on the dev network. PR #3145 provides a way to redirect comdb2db to comdb3db for the integration tier. However fdb would still fail: it would try to connect to the alpha nodes of the
remote database, which run on the prod network.

The pull request introduces a new tunable "remap_machine_class_to_fdb_tier", to remap a machine class to a fdb tier. For instance, `remap_machine_class_to_fdb_tier alpha integratione` would remap alpha to integration, such that cross-db queries against alpha would be redirected to integration. Similarly, `remap_machine_class_to_fdb_tier dev restore` would redirect cross-db queries against dev to restore.

The pull request also fixes a segfault from #3145.

(DRQS 168998941)

### CODE REVIEW

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ db/machclass.c @@
+ Please make sure that a mutex is unlocked before returning.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>0/0 tests failed</b> &#x26A0;.</p>

File: db/machclass.c [hunk 1]
```diff
@@ -55,6 +57,7 @@ int mach_class_init(void)
     Pthread_mutex_lock(&mach_mtx);
     classes = hash_init_strcaseptr(offsetof(struct machine_class, name));
     class_names = hash_init_i4(offsetof(struct machine_class, value));
+    class_names = hash_init_i4(offsetof(struct machine_class, value));
```
Comment (mponomar) [inline db/machclass.c:60]: Initialized twice?

Comment (riverszhang89) [reply db/machclass.c]: ugh .. corrected. Thanks for catching this!

**Review by mponomar — DISMISSED** [review]

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>3/429 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>socksql_master_swings<br>lostwrite<br>tunables</p>

**Review by mponomar — APPROVED** [review]

**Pull request merged**